lib/longrunning.js

```diff
@@ -0,0 +1,329 @@
+var events = require('events');
+var nextTick = require('process-nextick-args');
+var util = require('util');
+
+/**
+* A callback to upack a google.protobuf.Any message.
+* @callback anyDecoder
+* @param {google.protobuf.Any} message - The message to unpacked.
+* @return {Object} - The unpacked message.
+*/
+
+/**
+ * Describes the structure of a page-streaming call.
+ *
+ * @property {OperationsClient} operationsClient
+ * @property {anyDecoder} responseDecoder
+ * @property {anyDecoder} metadataDecoder
+ *
+ * @param {OperationsClient} operationsClient - The client used to poll or
+ *   cancel an operation.
+ * @param {anyDecoder=} responseDecoder - The decoder to unpack
+ *   the response message.
+ * @param {anyDecoder=} metadataDecoder - The decoder to unpack
+ *   the metadata message.
+ *
+ * @constructor
+ */
+function LongrunningDescriptor(
+    operationsClient, responseDecoder, metadataDecoder) {
+  this.operationsClient = operationsClient;
+  this.responseDecoder = responseDecoder;
+  this.metadataDecoder = metadataDecoder;
+}
+
+LongrunningDescriptor.prototype.apiCaller = function() {
+  return new LongrunningApiCaller(this);
+};
+
+exports.LongrunningDescriptor = LongrunningDescriptor;
+
+/**
+ * Creates an API caller that performs polling on a long running operation.
+ *
+ * @private
+ * @constructor
+ * @param {LongrunningDescriptor} longrunningDescriptor - Holds the
+ * decoders used for unpacking responses and the operationsClient
+ * used for polling the operation.
+ */
+function LongrunningApiCaller(longrunningDescriptor) {
+  NormalApiCaller.call(this);
+  this.longrunningDescriptor = longrunningDescriptor;
+}
+
+util.inherits(LongrunningApiCaller, NormalApiCaller);
+
+LongrunningApiCaller.prototype.call = function(
+    apiCall, argument, settings, canceller) {
+  canceller.call(this._wrapOperation.bind(this, apiCall, settings), argument);
+};
+
+LongrunningApiCaller.prototype._wrapOperation = function(
+  apiCall, settings, argument, callback) {
+  var backoffSettings = settings.longrunning;
+  if (!backoffSettings) {
+    backoffSettings =
+        createBackoffSettings(100, 1.3, 60000, null, null, null, 600000);
+  }
+  var operation;
+
+  var longrunningDescriptor = this.longrunningDescriptor;
+  return apiCall(argument, function(err, rawResponse) {
+    if (err) {
+      callback(err);
+      return;
+    }
+    operation = new Operation(
+      rawResponse,
+      longrunningDescriptor,
+      backoffSettings
+    );
+
+    callback(null, operation, rawResponse);
+  });
+};
+
+/**
+ * Wrapper for a google.longrunnung.Operation.
+ *
+ * @constructor
+ *
+ * @param {google.longrunning.Operation} grpcOp - The operation to be wrapped.
+ * @param {LongrunningDescriptor} longrunningDescriptor - This defines the
+ * operations service client and unpacking mechanisms for the operation.
+ * @param {BackoffSettings} backoffSettings - The backoff settings used in
+ * in polling the operation.
+ * @private
```

Comment:
I think Operation itself is not `private` (I think JSDoc will skip the documentation for the entire class if the constructor is marked as private).